packages/react/src/Dialog/Dialog.test.tsx

```diff
@@ -85,6 +85,25 @@ describe('Dialog', () => {
     expect(onClose).toHaveBeenCalledWith('escape')
   })
 
+  it('does not call `onClose` when click was not originated from backdrop', async () => {
+    const user = userEvent.setup()
+    const onClose = jest.fn()
+
+    const {getByRole} = render(<Dialog onClose={onClose}>Pay attention to me</Dialog>)
+
+    expect(onClose).not.toHaveBeenCalled()
+
+    const dialog = getByRole('dialog')
+    const backdrop = dialog.parentElement!
+
+    fireEvent.mouseDown(dialog)
+    fireEvent.mouseUp(backdrop)
+    // trigger the click on the backdrop, mouseUp doesn't do it for us
+    fireEvent.click(backdrop)
+
+    expect(onClose).not.toHaveBeenCalled()
+  })
+
```

Comment:
how do you feel about this @keithamus ? I do have to manually trigger the `click` event but it will fail if the current implementation were to be refactored out 🤔